replay/models/knn.py

```diff
@@ -82,6 +81,9 @@ def __init__(
         self.weighting = weighting
         self._nmslib_hnsw_params = nmslib_hnsw_params
 
+        if self._nmslib_hnsw_params:
+            super().__init__()
+
```

Comment:
Remove such constructor call. ANNMixin and its descendants should not have constructors at all.